File: kotlin-pr/50/kotlin/code-quality-naming/UserProfile.kt
class UserProfile {
    // {fact rule=code-quality-naming@v1.0 defects=0}
    private var processedData: MutableList<String> = mutableListOf()
    private var validItemCount: Int = 0
    private var hasValidItems: Boolean = false

    fun processStringArray(inputArray: List<String>) {
        processedData.clear()
        validItemCount = 0
        hasValidItems = false

        inputArray.forEach { inputItem ->
            if (inputItem.isNotEmpty()) {
                processedData.add(inputItem)
                validItemCount++
            }
        }

        hasValidItems = validItemCount > 0
    }
    // {/fact}
}

File: ruby-pr/53/ruby/code-quality-naming/user_profile.rb
class UserProfile
  # {fact rule=code-quality-naming@v1.0 defects=0}
  def initialize
    @processed_data = []
    @valid_item_count = 0
    @has_valid_items = false
  end

  def process_string_array(input_array)
    @processed_data = []
    @valid_item_count = 0
    @has_valid_items = false

    input_array.each do |input_item|
      unless input_item.empty?
        @processed_data << input_item
        @valid_item_count += 1
      end
    end

    @has_valid_items = @valid_item_count > 0
  end
  # {/fact}
end